vars/infra.groovy

```diff
@@ -141,8 +141,14 @@ Object withArtifactCachingProxy(boolean useArtifactCachingProxy = true, Closure
   if (useArtifactCachingProxy) {
     echo "INFO: using artifact caching proxy from '${requestedProxyProvider}' provider."
     configFileProvider(
-        [configFile(fileId: "artifact-caching-proxy-${requestedProxyProvider}", variable: 'MAVEN_SETTINGS')]) {
-          body()
+        [configFile(fileId: "artifact-caching-proxy-${requestedProxyProvider}", targetLocation: '/tmp/settings.xml')]) {
+          withEnv([
+            'MAVEN_SETTINGS=/tmp/settings.xml',
+            'MAVEN_ARGS=-s /tmp/settings.xml',
```

Comment:
re https://github.com/jenkins-infra/pipeline-library/pull/645#discussion_r1163250167
My initial version was like
```suggestion
        [configFile(fileId: "artifact-caching-proxy-${requestedProxyProvider}", variable: 'MAVEN_SETTINGS')]) {
          withEnv([
            "MAVEN_ARGS=-s $MAVEN_SETTINGS",
```
But when running the tests I saw only `MAVEN_ARGS=-s null` in the outputs. (Probably my mistake though)
I've also tested
```suggestion
        [configFile(fileId: "artifact-caching-proxy-${requestedProxyProvider}", variable: 'MAVEN_SETTINGS')]) {
          withEnv([
            "MAVEN_ARGS=-s $env.MAVEN_SETTINGS",
```
same `null` output.